aries/aries_vcx/src/handlers/out_of_band/sender.rs

```diff
@@ -128,10 +134,19 @@ impl OutOfBandSender {
         self.oob.clone().into()
     }
 
-    pub fn from_string(oob_data: &str) -> VcxResult<Self> {
-        Ok(Self {
-            oob: serde_json::from_str(oob_data)?,
-        })
+    pub fn to_json_string(&self) -> String {
+        self.to_aries_message().to_string()
+    }
+
+    pub fn to_base64_url(&self) -> String {
+        BASE64_URL_SAFE.encode(self.to_json_string())
```

Comment:
question; does base64 padding matter? it seems your encoding it _with_ padding, could result in a `=` at the end. If you don't want padding you can use `BASE64_URL_SAFE_NO_PAD`

Reply:
For parsing, we have to be able to handle with or without padding, but for encoding, it's not specified in RFC 0434 OOB (while it is in 0017 attachments) as far as I can see. The reason I switched to encoding _with_ padding was because the example base64url encoded invitation has a `=` at the end, indicating it was padded. I could take a look at the other frameworks to see what their implementations do here. Perhaps a clarification could be warranted to the RFC?

Reply:
similar comment to the receiver; do we think ppl want an API which returns this? or would they be happy with just "give me the invitation message" + "give my the invitation encoded as an OOB invitation"

Reply:
no worries from me then; the spec gives example with `=` as you said

Reply:
Can confirm that ACA-Py encodes with padding :+1:

Reply:
This did remind me to add a test to show we can parse with/without padding for base64url :+1: